airbyte-integrations/connectors/destination-langchain/destination_langchain/indexer.py

```diff
@@ -140,3 +141,37 @@ def check(self) -> Optional[str]:
         except Exception as e:
             return format_exception(e)
         return None
+
+
+class ChromaLocalIndexer(Indexer):
+    config: ChromaLocalIndexingModel
+
+    def __init__(self, config: ChromaLocalIndexingModel, embedder: Embedder):
+        super().__init__(config, embedder)
+
+    def _init_vectorstore(self):
+        self.vectorstore = Chroma(
+            collection_name=self.config.collection_name,
+            embedding_function=self.embedder.langchain_embeddings,
+            persist_directory=self.config.destination_path,
+        )
+
+    def pre_sync(self, catalog: ConfiguredAirbyteCatalog):
+        self._init_vectorstore()
+        for stream in catalog.streams:
+            if stream.destination_sync_mode == DestinationSyncMode.overwrite:
+                self.vectorstore._collection.delete(where={METADATA_STREAM_FIELD: {"$eq": stream.stream.name}})
+
+    def index(self, document_chunks, delete_ids):
+        for delete_in in delete_ids:
+            self.vectorstore._collection.delete(where={METADATA_RECORD_ID_FIELD: {"$eq": delete_in}})
+        self.vectorstore.add_documents(document_chunks)
```

Comment:
I might have misconfigured my connection, but my test sync failed on this line

```
  File "/airbyte/integration_code/destination_langchain/destination.py", line 65, in write
    batcher.flush()
  File "/airbyte/integration_code/destination_langchain/batcher.py", line 21, in flush
    self.flush_handler(list(self.buffer))
  File "/airbyte/integration_code/destination_langchain/destination.py", line 59, in <lambda></lambda>
    batcher = Batcher(BATCH_SIZE, lambda batch: self._process_batch(batch))
  File "/airbyte/integration_code/destination_langchain/destination.py", line 51, in _process_batch
    self.indexer.index(documents, ids_to_delete)
  File "/airbyte/integration_code/destination_langchain/indexer.py", line 168, in index
    self.vectorstore.add_documents(document_chunks)
  File "/usr/local/lib/python3.10/site-packages/langchain/vectorstores/base.py", line 104, in add_documents
    return self.add_texts(texts, metadatas, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/langchain/vectorstores/chroma.py", line 193, in add_texts
    self._collection.upsert(
  File "/usr/local/lib/python3.10/site-packages/chromadb/api/models/Collection.py", line 298, in upsert
    ids, embeddings, metadatas, documents = self._validate_embedding_set(
  File "/usr/local/lib/python3.10/site-packages/chromadb/api/models/Collection.py", line 357, in _validate_embedding_set
    validate_metadatas(maybe_cast_one_to_many(metadatas))
  File "/usr/local/lib/python3.10/site-packages/chromadb/api/types.py", line 169, in validate_metadatas
    validate_metadata(metadata)
  File "/usr/local/lib/python3.10/site-packages/chromadb/api/types.py", line 140, in validate_metadata
    raise ValueError(
ValueError: Expected metadata value to be a str, int, or float, got False which is a <class></class>
```

I used the Stripe source. Here is my destination config:
<img width="575" alt="Screen Shot 2023-07-25 at 11 43 05 AM" src="https://github.com/airbytehq/airbyte/assets/1451943/06789a8c-0cc2-422b-a09d-b157dafdbbc1">